mapie/quantile_regression.py

```diff
@@ -0,0 +1,414 @@
+            intervals. Alpha is the complement of the target coverage level.
+            Only used at prediction time. By default 0.1
+
+        Returns
+        -------
+        ArrayLike
+            An ArrayLike of three values, first one being the lower quantile
+            value, second the upper quantile value and final being the quantile
+            at which the prediction will be made.
+
+        Raises
+        ------
+        ValueError
+            If alpha is not a float.
+        ValueError
+            If the value of alpha is not between 0 and 0.5.
+        """
+        if isinstance(alpha, float):
+            if np.any(np.logical_or(alpha <= 0, alpha >= 0.5)):
+                raise ValueError(
+                    "Invalid alpha. Allowed values are between 0 and 0.5."
+                )
+            else:
+                alpha_np = np.array([alpha / 2, 1 - alpha / 2, 0.5])
+        else:
+            raise ValueError(
+                "Invalid alpha. Allowed values are float."
+            )
+        return alpha_np
+
+    def _check_estimator(
+        self, estimator: Optional[RegressorMixin] = None
+    ) -> RegressorMixin:
+        """
+        Perform several checks on the estimator to check if it has
+        all the required specifications to be used with this methodology.
+
+        Parameters
+        ----------
+        estimator : Optional[RegressorMixin], optional
+            Estimator to check, by default ``None``.
+
+        Returns
+        -------
+        RegressorMixin
+            The estimator itself or a default ``QuantileRegressor`` instance
+            with ``solver`` set to "highs".
+
+        Raises
+        ------
+        ValueError
+            If the estimator fit or predict methods.
+        ValueError
+            We check if it's a known estimator that does quantile regression
+            according to the dictionnary set quantile_estimator_params.
+            This dictionnary will need to be updated with the latest new
+            available estimators.
+        ValueError
+            The estimator does not have the "loss_name" in its parameters and
+            therefore can not be used as an estimator.
+        ValueError
+            There is no quantile "loss_name" and therefore this estimator
+            can not be used as a ``MapieQuantileRegressor``.
+        ValueError
+            The parameter to set the alpha value does not exist in this
+            estimator and therefore we cannot use it.
+        """
+        if estimator is None:
+            return QuantileRegressor(solver="highs")
+        if not (hasattr(estimator, "fit") and hasattr(estimator, "predict")):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        name_estimator = estimator.__class__.__name__
+        if name_estimator == "QuantileRegressor":
```

Comment:
Since a dictionary is built for all methods, do we need to distinguish `QuantileRegressor` from other methods ?